src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String LINE = " - ";
+    private static final String REPORT_HEADER_FORMAT = "Report for period %s - %s%n";
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom, DATE_FORMAT);
+        LocalDate toDate = LocalDate.parse(dateTo, DATE_FORMAT);
+        int[] salaries = new int[names.length];
+        for (String record : data) {
+            String[] recordData = record.split(" ");
+            LocalDate recordDate = LocalDate.parse(recordData[0], DATE_FORMAT);
+            if (!recordDate.isBefore(fromDate) && !recordDate.isAfter(toDate)) {
+                String name = recordData[1];
```

Comment:
Add constants for the indexes 0-1-2-3
```suggestion
                String name = recordData[NAME_INDEX];
```